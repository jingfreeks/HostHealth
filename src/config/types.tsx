export type PcitiesStateProps = {
  data?: PopularListDetails[];
  loading: boolean;
};

export type PopularListDetails = {
  image: string;
  city: string;
  state: string;
  matches: string;
  salary: string;
};

export type SuggestedCardDetailsProps = {
  image?: string;
  city?: string;
  state?: string;
  jobtitle?: string;
  company?: string;
  dept?: string;
  weeks?: string;
  shift?: string;
  match?: string;
  salaryrange?: string;
  address?: string;
  joborderno?: string;
  id?: number;
};

export type SuggestedJobSateProps = {
  data?: SuggestedCardDetailsProps[];
  loading: boolean;
};

export type State = {
  pcities: PcitiesStateProps;
  suggetedjob: SuggestedJobSateProps;
};
