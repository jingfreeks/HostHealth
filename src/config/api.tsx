import {createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react';

export const healtHostApi = createApi({
  reducerPath: 'healtHostApi',
  baseQuery: fetchBaseQuery({baseUrl: 'https://run.mocky.io/v3/'}),
  endpoints: builder => ({
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    getPopularCities: builder.query<any, any>({
      query: () => '5e58bb33-dcf4-4d68-8a64-c68775955769',
    }),
    getSuggestedJobs: builder.query({
      query: () => 'd96eff10-e889-47d3-b8ad-3b2a106b71cd',
    }),
  }),
});

export const {useGetPopularCitiesQuery, useGetSuggestedJobsQuery} =
  healtHostApi;
