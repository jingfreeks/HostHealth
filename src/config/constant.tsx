export const message={
    100001:'Token Expired you must logout and login it again',
    100002:'No state matches your preferred locations',
    100003:'No cities matches your preferred locations',
    100004:'No Department matches your preferred locations',
    100005:'Department Name Should be required',
    100006:'Shift Name Should be required',
    100007:'No Shift matches your preferred locations',
    100008:'No Company matches your preferred locations',
    100009:'City Name Should be required',
    100010:'Company Name Should be required',
    100011:'Address Should be required',
    100012:'State should be required',
    100013:'Bank Name should be required',
    100014:'Bank Address shoud be required',
    100015:'No Banks matches your preferred locations'
}

export const TagTypes=['City', 'Jobs', 'MyJobs', 'JobDetails', 'Profile','States','Department','Shift','Company','Banks']