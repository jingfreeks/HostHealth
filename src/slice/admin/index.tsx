export {
    useGetShiftQuery,
    useAddShiftMutation,
    useUpdateShiftMutation,
    useDeleteShiftMutation,
  } from './shift';
  export {
    useGetDeptQuery,
    useAddDeptMutation,
    useUpdateDeptMutation,
    useDeleteDeptMutation,
  } from './department';
  export {
    useGetCompanyQuery,
    useAddCompanyMutation,
    useUpdateCompanyMutation,
    useDeleteCompanyMutation,
  }from './company'