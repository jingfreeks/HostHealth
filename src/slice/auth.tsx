import {createSlice} from '@reduxjs/toolkit';

const authSlice = createSlice({
  name: 'auth',
  initialState: {user: null, token: null,refreshToken:null,userId:null,roles:[],onBoarding:false},
  reducers: {
    setCredentials: (state, action) => {
      const {user, accessToken,userId,roles,isOnBoarding,refreshToken} = action.payload;
      state.user = user;
      state.token = accessToken;
      state.refreshToken=refreshToken
      state.userId = userId;
      state.roles=roles;
      state.onBoarding=isOnBoarding;
    },
    setLogout: (state) => {
      state.user = null;
      state.token = null;
      state.userId = null;
      state.refreshToken=null;
      state.roles=[]
      state.onBoarding=false;
    },
  },
});


export const {setCredentials,setLogout}=authSlice.actions

export default authSlice.reducer

export const selectCurrentUser = (state:any)=>state.auth.user
export const selectCurrentUserId=(state:any)=>state.auth.userId
export const selectCurrentToken = (state:any)=>state.auth.token
export const selectCurrentAuth=(state:any)=>state.auth
export const selectUserRoles=(state:any)=>state.auth.roles