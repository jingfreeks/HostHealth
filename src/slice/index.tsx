export {
  useGetShiftQuery,
  useAddShiftMutation,
  useUpdateShiftMutation,
  useDeleteShiftMutation,
  useGetDeptQuery,
  useAddDeptMutation,
  useUpdateDeptMutation,
  useDeleteDeptMutation,
  useGetCompanyQuery,
  useAddCompanyMutation,
  useUpdateCompanyMutation,
  useDeleteCompanyMutation,
  useGetBanksQuery,
  useAddBanksMutation,
  useUpdateBanksMutation,
  useDeleteBanksMutation,
} from './admin';

export {
  useGetCityQuery,
  useAddCityMutation,
  useUpdateCityMutation,
  useDeleteCityMutation,
} from './city';

export {
  useGetProfileQuery,
  useUpdateProfileMutation,
  useUploadProfileMutation,
} from './profile';

export {
  useGetStateQuery,
  useAddStatesMutation,
  useUpdateStatesMutation,
  useDeleteStatesMutation,
} from './state';

export {
  useGetOnBoardingProfileQuery,
  useUpdateOnBoardingProfileMutation,
  useGetOnBoardingBankInfoQuery,
  useUpdateOnBoardingBankInfoMutation,
} from './onboarding';

export {useGetJobsQuery,useGetCityJobsQuery} from './suggested'