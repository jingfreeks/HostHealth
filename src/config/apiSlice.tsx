import {setCredentials, setLogout} from '@/slice/auth';
import Config from 'react-native-config';
import {TagTypes} from './constant';
import {createApi, fetchBaseQuery} from '@reduxjs/toolkit/query/react';

const baseQuery = fetchBaseQuery({
  baseUrl: Config.DEV_BASE_URL,
  credentials: 'include',
  prepareHeaders: (headers, {getState}: any) => {
    const token = getState().auth.token;
    if (token) {
      headers.set('authorization', `Bearer ${token}`);
    }
    return headers;
  },
});

const baseQueryWithAuth = async (args: any, api: any, extraOptions: string) => {
  let result: any =
    args?.url === '/logout'
      ? await baseQuery(
          {
            url: '/auth/logout',
            method: 'POST',
            headers: {
              Cookie: `jwt=${api.getState().auth.refreshToken}`,
            },
          },
          api,
          extraOptions,
        )
      : await baseQuery(args, api, extraOptions);
  if (result?.error?.status === 403) {
    const refereshResult = await baseQuery(
      {
        url: '/auth/refresh',
        headers: {
          Cookie: `jwt=${api.getState().auth.refreshToken}`,
        },
      },
      api,
      extraOptions,
    );
    if (refereshResult?.data) {
      const user = api.getState().auth.user;
      const roles = api.getState().auth.roles;
      const userId = api.getState().auth.userId;
      const refreshToken = api.getState().auth.refreshToken;
      const isOnBoarding = api.getState().auth.onBoarding
      api.dispatch(
        setCredentials({...refereshResult.data, user, roles, refreshToken,userId,isOnBoarding}),
      );

      result = await baseQuery(args, api, extraOptions);
    }
  }
  return result;
};

export const apiSlice = createApi<any, any>({
  baseQuery: baseQueryWithAuth,
  reducerPath: 'api',
  endpoints: builder => ({}),
  tagTypes: TagTypes as string[] & undefined,
});
