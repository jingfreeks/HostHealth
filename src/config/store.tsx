import {configureStore, combineReducers} from '@reduxjs/toolkit';
import type { PreloadedState } from '@reduxjs/toolkit'
import {setupListeners} from '@reduxjs/toolkit/query';
import {healtHostApi} from './api';
// import pcitiesslice from '@/slice/pcities';
import loginslice from '@/slice/login';
import suggestedjobs from '@/slice/suggested';
import {apiSlice} from './apiSlice';
import authReducer from '@/slice/auth';

const rootReducer = combineReducers({
  // [healtHostApi.reducerPath]: healtHostApi.reducer,
  [apiSlice.reducerPath]: apiSlice.reducer,
  auth: authReducer,
  // pcities: pcitiesslice,
  suggetedjob: suggestedjobs,
  login: loginslice,
});


export const setupstore = (preloadedState?:PreloadedState<RootState>)=>configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware().concat(apiSlice.middleware),
  devTools: true,
  preloadedState
});

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupstore>;
export type AppDispatch = AppStore['dispatch'];

export const store =setupstore({});
setupListeners(store.dispatch);
