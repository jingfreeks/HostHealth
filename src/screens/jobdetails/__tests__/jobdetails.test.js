import React from 'react';
import {renderWithProviders} from '@/utils/testframeworknew';
import JobDetails from '../jobdetails';
import {setCredentials} from '@/slice/auth'
import {setupstore} from '@/config/store';

describe('Job Details Screen', () => {
  const props = {
    route: {
      params: {
        jobdetail: {
          image: '',
          address: '',
          jobtitle: '',
          joborderno: '',
          company: '',
          salaryrange: '',
          shift: '',
        },
      },
    },
  };
  it('Should work as expected to get snapshot', () => {
    const store = setupstore();
    store.dispatch(
      setCredentials({
        user: 'testing',
        accessToken: 'testAccesstoken',
        userId: '123344',
        roles: ['Employee'],
        onBoarding: false,
      }),
    );
    const all = renderWithProviders(<JobDetails {...props} />,{store});
    expect(all.toJSON()).toMatchSnapshot();
  });

});
