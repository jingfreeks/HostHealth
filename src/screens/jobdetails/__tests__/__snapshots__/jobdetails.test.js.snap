// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Details Screen Should work as expected to get snapshot 1`] = `
<ActivityIndicator
  color="#0000ff"
  size="small"
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 19.61764705882353,
        "paddingLeft": 19.61764705882353,
        "paddingRight": 19.61764705882353,
        "paddingTop": 19.61764705882353,
      },
    ]
  }
/>
`;
